src/ElmFormat/Render/Box.hs

```diff
@@ -1595,6 +1547,112 @@ formatExpression' elmVersion importInfo context aexpr =
+            if'
+
+        cond' =
+            formatIfOpening (line $ keyword "if") $
+                formatCommentedExpression elmVersion importInfo SyntaxSeparated cond
+
+        then' =
+            concat $
+                [ Maybe.maybeToList $ formatComments preThen
+                , [ formatExpression elmVersion importInfo SyntaxSeparated thenExpr ]
+                , Maybe.maybeToList $ formatComments postThen
+                ]
+
+        else' =
+            (Maybe.maybeToList $ formatComments elsComments)
+                ++ [ formatExpression elmVersion importInfo SyntaxSeparated els ]
+    in
+    case
+        ( multiline
+        , isLine cond'
+        , allSingles then'
+        , map (formatElseIf elmVersion importInfo) elseifs
+        , allSingles else'
+        )
+    of
+        (AST.JoinAll, Right singleCond, Right singleThens, [], Right singleElses)  ->
+            line $ row $ concat
+                [ [singleCond, space]
+                , List.intersperse space singleThens
+                , [ space, keyword "else", space ]
+                , List.intersperse space singleElses
+                ]
+
+        (_, ifCond, thenBody, elseIfs', elseBody) ->
+            stack1
+                [ boxOrLineToBox ifCond
+                , indent $ boxesOrLinesToBox True thenBody
+                ]
+                |> andThen elseIfs'
+                |> andThen
+                    [ blankLine
+                    , line $ keyword "else"
+                    , indent $ boxesOrLinesToBox True elseBody
+                    ]
+
+
+formatIfOpening :: Box -> Box -> Box
+formatIfOpening key cond =
+    case (key, cond) of
+        (SingleLine key', SingleLine cond') ->
+            line $ row [ key', space, cond', space, keyword "then" ]
+
+        _ ->
+            stack1
+                [ key
+                , cond |> indent
+                , line $ keyword "then"
+                ]
+
+
+formatElseIf :: ElmVersion -> ImportInfo -> (AST.Comments, AST.Expression.IfClause) -> Box
+formatElseIf elmVersion importInfo (ifComments, (cond', body')) =
+    let
+        key =
+            case (formatHeadCommented id (ifComments, line $ keyword "if")) of
+                SingleLine key' ->
+                    line $ row [ keyword "else", space, key' ]
+                key' ->
+                    stack1
+                        [ line $ keyword "else"
+                        , key'
+                        ]
+    in
+    stack1
+        [ blankLine
+        , formatIfOpening key $ formatCommentedExpression elmVersion importInfo SyntaxSeparated cond'
+        , indent $ formatCommented_ True (formatExpression elmVersion importInfo SyntaxSeparated) body'
+        ]
+
+
+boxOrLineToBox :: Either Box Line -> Box
+boxOrLineToBox boxOrLine =
+    case boxOrLine of
+        Right singleLine -> line singleLine
+        Left box -> box
+
+
+boxesOrLinesToBox :: Bool -> Either [Box] [Line] -> Box
```

Comment:
Maybe we could put this function and `boxOrLineToBox` to `Box.hs`?
I'm not sure because of the use of `ElmStructure.forceableSpaceSepOrStack1`.